packages/widgets/address-update/src/modules/addressUpdate/sagas.js

```diff
@@ -0,0 +1,126 @@
+import {actions as formActions, isValid as isValidSelector} from 'redux-form'
+import {all, call, put, select, takeEvery, takeLatest} from 'redux-saga/effects'
+import {form, rest, display, notification} from 'tocco-app-extensions'
+import {api} from 'tocco-util'
+
+import {fetchAddress, updateAddress} from '../../util/api/entities'
+import * as actions from './actions'
+
+export const addressUpdateSelector = state => state.addressUpdate
+
+export const inputSelector = state => state.input
+
+export const FORM_ID = 'addressForm'
+
+export const formSagaConfig = {
+  formId: FORM_ID,
+  stateSelector: addressUpdateSelector
+}
+
+export default function* sagas() {
+  yield all([
+    takeLatest(actions.LOAD_VIEW, loadView),
+    takeLatest(actions.UNLOAD_VIEW, unloadView),
+    takeLatest(actions.TOUCH_ALL_FIELDS, form.sagasUtils.touchAllFields, formSagaConfig),
+    takeEvery(actions.SUBMIT_FORM, submitForm),
+    takeEvery(actions.FIRE_TOUCHED, fireTouched)
+  ])
+}
+
+export function* loadDetailFormDefinition(formName, mode) {
+  const formDefinition = yield call(rest.fetchForm, formName, mode)
+  yield put(actions.setFormDefinition(formDefinition))
+  const fieldDefinitions = yield call(form.getFieldDefinitions, formDefinition)
+  yield put(actions.setFieldDefinitions(fieldDefinitions))
+  return {formDefinition, fieldDefinitions}
+}
+
+export function* loadEntity(fieldDefinitions) {
+  const {
+    appContext: {widgetConfigKey}
+  } = yield select(inputSelector)
+  const paths = yield call(form.getUsedPaths, fieldDefinitions)
+  const entity = yield call(fetchAddress, widgetConfigKey, {paths})
+  yield put(actions.setEntity(entity))
+  return entity
+}
+
+export function* unloadView() {
+  yield put(actions.setEntity(null))
+  yield put(formActions.destroy(FORM_ID))
+}
+
+export function* loadView() {
+  const {formName} = yield select(inputSelector)
+  yield call(loadDetailFormDefinition, formName, 'detail')
+  yield call(loadData)
+}
+
+export function* updateFormSubmit(entity, fieldDefinitions) {
+  const {
+    appContext: {widgetConfigKey}
+  } = yield select(inputSelector)
+  const updateResponse = yield call(updateAddress, widgetConfigKey, entity, fieldDefinitions)
+  if (updateResponse.status === 404) {
+    // record was most likely moved to another business unit
```

Comment:
shouldn't we show a toaster here?

Reply:
👍  removed `404` as `acceptedStatusCodes` in the `updateAddress` fn. so there should be an error toaster now.